lib/docker/synchronizer.go

```diff
@@ -0,0 +1,204 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/docker/distribution"
+	dockerref "github.com/docker/distribution/reference"
+	regclient "github.com/docker/distribution/registry/client"
+	dockerapi "github.com/fsouza/go-dockerclient"
+	"github.com/gravitational/gravity/lib/loc"
```

Comment:
nit: move these 3 to a separate group (local imports) as it is customary elsewhere:
```go
import (
	"context"
	"fmt"
	"strings"

	"github.com/gravitational/gravity/lib/loc"
	"github.com/gravitational/gravity/lib/run"
	"github.com/gravitational/gravity/lib/utils"

	"github.com/docker/distribution"
	dockerref "github.com/docker/distribution/reference"
	regclient "github.com/docker/distribution/registry/client"
	dockerapi "github.com/fsouza/go-dockerclient"
	"github.com/gravitational/trace"
	log "github.com/sirupsen/logrus"
)
```